switchlink/switchlink_route_test.cc

```diff
@@ -0,0 +1,738 @@
+#include <arpa/inet.h>
+#include <memory.h>
+#include <vector>
+
+#include "netlink/msg.h"
+#include "gtest/gtest.h"
+
+extern "C" {
+#include "switchlink/switchlink_handle.h"
+#include "switchlink/switchlink_int.h"
+#include "switchlink/switchlink_route.h"
+}
+
+using namespace std;
+
+switchlink_handle_t g_default_vrf_h = 2;
+switchlink_handle_t g_cpu_rx_nhop_h = 3;
+
+#define IPV4_ADDR(a, b, c, d) (((a) << 24) | ((b) << 16) | ((c) << 8) | (d))
```

Comment:
We are using `inet_pton()` for IPv6. Why not use the same for IPv4 address creation instead of this format?

Reply:
for ipv4 address we use nla_put_u32() instead of nla_put() which is used for ipv6. nla_put() expects ipv6 address in struct in6_addr, whereas for ipv4 we are only interested in putting the s_addr field of in_addr structure which is a 32 bit int.